.github/workflows/publish.yaml

```diff
@@ -0,0 +1,40 @@
+# Flow that has 2 possible logics depending on the event:
+#   - "pull_request":
+#     - `cargo publish --dry-run`: Execute a dry run for publication on each crate in the repo, making sure
+#     that publication will be possible on merge
+#   - "push":
+#   - `cargo publish --token ***`: For every crate in our workspace, publish it on crates.io
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+name: Publish
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
```

Comment:
`actions-rs` has been [unmaintained for years now](https://github.com/actions-rs/toolchain/issues/216). Can we use [`dtolnay/rust-toolchain`](https://github.com/dtolnay/rust-toolchain) instead?

Reply:
thanks for the good point joão.

impressive how libraries can have a lasting mark in the open documentation and still suffer from a maintenance problem

Reply:
Sorry Philippe, I just noticed now that if we merge this PR and then merge the others it will publish `kythera`, same with every PR to the main that we need in the future (like doc, CI changes etc). Should we make this [manual](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) instead?

Reply:
Is it really a problem that `kythera` is published on every **push** to `main`? To me, it is kind of the expected behavior 🙃🤔.

> ps: btw note that the base branch for his pr has changed to `develop`

Reply:
thanks for the quick sync joão. we will thus move forward on these matters by

- publishing all crates (even possibly untouched ones) at the same pace, using a common version number imposed to all crates in the workspace
- merging our new deployment workflow with your previous work on the matrix of crates 👍